src/transaction.ts

```diff
@@ -706,31 +717,57 @@ export class Snapshot extends EventEmitter {
       });
     };
 
-    return partialResultStream(this._wrapWithIdWaiter(makeRequest), {
-      json,
-      jsonOptions,
-      maxResumeRetries,
-      columnsMetadata,
-      gaxOptions,
-    })
-      ?.on('response', response => {
-        if (response.metadata && response.metadata!.transaction && !this.id) {
-          this._update(response.metadata!.transaction);
-        }
+    const q = {tableName: table, opts: this._observabilityOptions};
+    return startTrace('Snapshot.createReadStream', q, span => {
+      const prs = partialResultStream(this._wrapWithIdWaiter(makeRequest), {
+        json,
+        jsonOptions,
+        maxResumeRetries,
+        columnsMetadata,
+        gaxOptions,
       })
-      .on('error', err => {
-        const isServiceError = err && typeof err === 'object' && 'code' in err;
-        if (
-          !this.id &&
-          this._useInRunner &&
-          !(
-            isServiceError &&
-            (err as grpc.ServiceError).code === grpc.status.ABORTED
-          )
-        ) {
-          this.begin();
-        }
-      });
+        ?.on('response', response => {
+          if (response.metadata && response.metadata!.transaction && !this.id) {
+            this._update(response.metadata!.transaction);
+          }
+        })
+        .on('error', err => {
+          const isServiceError =
+            err && typeof err === 'object' && 'code' in err;
+          if (
+            !this.id &&
+            this._useInRunner &&
+            !(
+              isServiceError &&
+              (err as grpc.ServiceError).code === grpc.status.ABORTED
+            )
+          ) {
+            this.begin();
+          }
+          setSpanError(span, err);
+        })
+        .on('end', err => {
+          if (err) {
+            setSpanError(span, err);
+          }
+          span.end();
+        });
+
+      if (!prs) {
+        return prs;
+      }
+
+      if (prs instanceof Stream) {
```

Comment:
prs should always be stream

Reply:
In tests there were some cases in which it wasn't.

Reply:
@surbhigarg92 these tests been failing without this check

```shell
Transaction
       Snapshot
         createReadStream
```

producing

```shell
1) Transaction
       Snapshot
         createReadStream
           should send the correct request:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:270:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  2) Transaction
       Snapshot
         createReadStream
           should use the transaction id if present:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:284:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  3) Transaction
       Snapshot
         createReadStream
           should configure `singleUse` if id is absent:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:296:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  4) Transaction
       Snapshot
         createReadStream
           should set request tag:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:307:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  5) Transaction
       Snapshot
         createReadStream
           should send the correct `reqOpts`:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:342:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  6) Transaction
       Snapshot
         createReadStream
           should pass along `gaxOpts`:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received undefined
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:352:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  7) Transaction
       Snapshot
         createReadStream
           should pass a stream to `PartialResultStream`:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:364:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  8) Transaction
       Snapshot
         createReadStream
           should update the `resumeToken` for subsequent requests:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:376:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  9) Transaction
       Snapshot
         createReadStream
           should return a `PartialResultStream`:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:388:33)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  10) Transaction
       Snapshot
         createReadStream
           should pass along row options:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:405:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  11) Transaction
       Snapshot
         createReadStream
           should accept directedReadOptions set for client:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:436:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  12) Transaction
       Snapshot
         createReadStream
           should override directedReadOptions set at client level when passed at request level:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Snapshot.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:475:18)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  13) Transaction
       Transaction
         createReadStream
           should send the correct options:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Transaction.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:2221:21)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  14) Transaction
       Transaction
         createReadStream
           should set transaction tag if not `singleUse`:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Transaction.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:2242:21)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)

  15) Transaction
       Transaction
         createReadStream
           should set transaction tag if `begin`:
     TypeError [ERR_INVALID_ARG_TYPE]: The "stream" argument must be an instance of ReadableStream, WritableStream, or Stream. Received an instance of EventEmitter
      at eos (node:internal/streams/end-of-stream:74:11)
      at <anonymous> (src/transaction.ts:759:17)
      at <anonymous> (src/instrument.ts:145:16)
      at AsyncLocalStorage.run (node:async_hooks:346:14)
      at AsyncLocalStorageContextManager.with (node_modules/@opentelemetry/context-async-hooks/src/AsyncLocalStorageContextManager.ts:40:36)
      at ContextAPI.with (node_modules/@opentelemetry/api/src/api/context.ts:77:42)
      at Tracer.startActiveSpan (node_modules/@opentelemetry/sdk-trace-base/src/Tracer.ts:241:28)
      at startTrace (src/instrument.ts:114:49)
      at Transaction.createReadStream (src/transaction.ts:721:22)
      at Context.<anonymous> (test/transaction.ts:2256:21)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)
```